apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableVersionRange.cs

```diff
@@ -0,0 +1,97 @@
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersionRange"/>.
+        /// </summary>
+        /// <param name="start">Inclusive lower bound. If null, interpreted as 0.</param>
+        /// <param name="end">Exclusive upper bound. If null, interpreted as infinity.</param>
+        public BigtableVersionRange(BigtableVersion? start, BigtableVersion? end)
+        {
+            Start = start;
+            End = end;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersionRange"/>.
+        /// </summary>
+        /// <param name="start">
+        /// Inclusive lower bound non-negative version value, or -1 to initialize from the
+        /// microseconds of DateTime.UtcNow. If null, interpreted as 0.
+        /// </param>
+        /// <param name="end">
+        /// Exclusive upper bound non-negative version value, or -1 to initialize from the
+        /// microseconds of DateTime.UtcNow. If null, interpreted as infinity.
+        /// </param>
+        public BigtableVersionRange(long? start, long? end)
+            : this(start.ToVersion(), end.ToVersion()) { }
+
+        // TODO: Get clarification on use cases to see how useful this would be.
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersionRange"/>.
+        /// </summary>
+        /// <param name="startTimestamp">
+        /// Inclusive lower bound timestamp whose microseconds should be used as the version value.
+        /// If null, interpreted as 0.  It must be specified in UTC.
+        /// </param>
+        /// <param name="endTimestamp">
+        /// Exclusive upper bound timestamp whose microseconds should be used as the version value.
+        /// If null, interpreted as infinity.  It must be specified in UTC.
+        /// </param>
+        public BigtableVersionRange(DateTime? startTimestamp, DateTime? endTimestamp)
+            : this(startTimestamp.ToVersion(), endTimestamp.ToVersion()) { }
+
+        /// <summary>
+        /// Gets the inclusive lower bound. If null, interpreted as 0.
+        /// </summary>
+        public BigtableVersion? Start { get; }
+
+        /// <summary>
+        /// Gets the exclusive upper bound. If null, interpreted as infinity.
+        /// </summary>
+        public BigtableVersion? End { get; }
+
+        /// <inheritdoc />
+        public bool Equals(BigtableVersionRange other) =>
+            other != null && Start == other.Start && End == other.End;
+
+        /// <inheritdoc />
+        public override bool Equals(object obj) => Equals(obj as BigtableVersionRange);
+
+        /// <inheritdoc />
+        public override int GetHashCode() =>
+            Start.HasValue.GetHashCode() ^ End.HasValue.GetHashCode() ^
+            Start.GetHashCode() ^ End.GetHashCode();
+
+        /// <inheritdoc />
+        public override string ToString() =>
+            $"{nameof(BigtableVersionRange)}: Start={Start?.Value.ToString() ?? " "}; End={End?.Value.ToString() ?? " "}";
```

Comment:
Consider "(none)" or "<>" as an alternative to " " for a missing value?

Reply:
Done (using "(none)")